lib/puppet/provider/nameservice.rb

```diff
@@ -141,7 +143,8 @@ def self.autogen_id(field, resource_type)
     when :user;   database = :passwd;  method = :uid
     when :group;  database = :group;   method = :gid
     else
-      raise Puppet::DevError, "Invalid resource name #{resource}"
+      #TRANSLATORS "autogen_id()" is a method name sna should not be translated
+      raise Puppet::DevError, _("autogen_id() does not support auto genration of id for resource type %{resource_type}") % { resource_type: resource_type }
```

Comment:
This is a message change.
We don't see this, as this method is only called by autogen and only when resource_type is set to :user or :group. In any case the message is now more descriptive of what the issue is and its printing the right variable

Reply:
Did you mean "and" instead of "sna"?  Also, "generation" instead of "genration".